daemon/policy.go

```diff
@@ -282,6 +308,45 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (uint64, err
 	return rev, nil
 }
 
+// ReactToRuleUpdates waits until wg is complete to do the following
+// * regenerate all endpoints in epsToRegen
+// * bump the policy revision of all endpoints not in epsToRegen, but which are
+//   in allEps, to revision rev.
+func (d *Daemon) ReactToRuleUpdates(wg *sync.WaitGroup, allEps []policy.IdentityConsumer, epsToRegen *policy.IDSet, rev uint64) {
+	// Wait until we have calculated which endpoints need to be selected
+	// across multiple goroutines.
+	wg.Wait()
+
+	epsToRegen.Mutex.RLock()
+	defer epsToRegen.Mutex.RUnlock()
+
+	// If an endpoint is not in the list of endpoints to regenerate, then the
+	// policy revision for the endpoint needs to be bumped so we can reflect that
+	// said endpoint realizes the state of the policy repository at revision rev.
+	endpointsToBumpRevision := map[uint16]struct{}{}
+	for _, ep := range allEps {
+		if ep == nil {
+			continue
+		}
+		epID := ep.GetID16()
+		if _, ok := epsToRegen.IDs[epID]; !ok {
+			endpointsToBumpRevision[epID] = struct{}{}
```

Comment:
and the `PolicyRevisionBumpEvent` interface to the `policy.IdentityConsumer` and in here, do `ep.PolicyRevisionBumpEvent(rev)` directly.

Reply:
I don't want to do it directly because in the current implementation of the EventQueue, the bumping of the revision may block due to events currently being processed on the queue. I can definitely add the function to the `IdentityConsumer` interface, though, as I agree that that makes things cleaner.

Reply:
Let me rewrite
**add** the `PolicyRevisionBumpEvent` **function** to the `policy.IdentityConsumer` and in here, do `ep.PolicyRevisionBumpEvent(rev)` directly.